CMakeLists.txt

```diff
@@ -5,11 +5,52 @@ project(blueprint VERSION 0.1.0)
 # (practical to debug the plugin processor directly from your IDE)
 option(JUCE_BUILD_EXTRAS "Build JUCE Extras" OFF)
 
+set(REACTJUCE_JS_LIBRARY HERMES CACHE STRING "The JS Engine to use: either HERMES or DUKTAPE")
+
 add_subdirectory(ext/juce)
 
 # Adding any custom modules you might have:
 juce_add_module(react_juce)
 
+# Setup the JS engine/interpreter to use.
+if (REACTJUCE_JS_LIBRARY STREQUAL "HERMES")
+    #TODO: Review Question:
```

Comment:
@nick-thompson, TODO here worth discussing.

Reply:
Also, this is the new approach I imagined for making the JS library/engine use easily switchable by the user. Doing this all I need to do is change one variable in the cmake cache for `REACTJUCE_JS_LIBRARY` and everything "just works"

Reply:
Yep, love this approach, I think the only other thing would be to add another var up top that specifies the path to hermes, allows a user override, and defaults to react_juce/hermes, right?

Reply:
Totally. I'm gonna experiment with this just to make sure we don't annihilate include paths via `juce_add_module` etc. if we follow suit with Duktape path.

Reply:
@nick-thompson , before I close this. Do we think a custom hermes include path is actually necessary ? I think this is only a thing we really require for Duktape. If user's are using some custom hermes checkout that's fairly non-standard and in which case I'd argue it would make more sense they had a blueprint fork and modified submodule.

Reply:
Well, I feel like if I were in the situation trying to use blueprint but needing a custom hermes (or qjs, or duktape) and _needed_ to fork to achieve that, it would be annoying. But, AFAIK we have 0 users with that use case, so yea, good point, let's hold back any real efforts on this until it becomes apparent people want/need it

Reply:
Yeah. OK I'll follow this PR up with another for "custom library includes". I guess my reservation is that we submodule to be locked against a particular version for API compatibility etc. etc. The same could be argued with customised Duktape though. I'd be tempted to get Hermes and QuickJS in and then add a step for all three libs where the user can (as much as possible) take the same basic configure approaches for each